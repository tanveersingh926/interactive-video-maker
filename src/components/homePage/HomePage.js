import React from "react";
import addVideoIcon from "../../assets/img/add-video-icon.svg";
import {
  Card,
  CardImg,
  CardBody,
  CardTitle,
  CardText,
  Row,
  Col
} from "reactstrap";
import { Link } from "react-router-dom";

const HomePage = () => {
  return (
    <Row>
      <Col sm="3" className="mt-4">
        <Card>
          <CardImg top width="100%" src={addVideoIcon} alt="Add New Video 1" />
          <CardBody>
            <CardTitle>New Video</CardTitle>
            <CardText>
              Lets make another with interactive and fun to watch.
            </CardText>
            <Link className="btn btn-primary" to={"/add-new-video"}>
              Lets go!
            </Link>
          </CardBody>
        </Card>
      </Col>
    </Row>
  );
};

export default HomePage;
