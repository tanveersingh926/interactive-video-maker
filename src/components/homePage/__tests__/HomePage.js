import React from "react";
import { render, cleanup } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import HomePage from "../HomePage";

describe("component should render with all basic sections", () => {
  afterEach(cleanup);

  it("should render page snapshot", async () => {
    const { asFragment } = render(
      <BrowserRouter>
        <HomePage />
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
