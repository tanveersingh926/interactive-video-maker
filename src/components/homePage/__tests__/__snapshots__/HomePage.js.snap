// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component should render with all basic sections should render page snapshot 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="mt-4 col-sm-3"
    >
      <div
        class="card"
      >
        <img
          alt="Add New Video 1"
          class="card-img-top"
          src="add-video-icon.svg"
          width="100%"
        />
        <div
          class="card-body"
        >
          <div
            class="card-title"
          >
            New Video
          </div>
          <p
            class="card-text"
          >
            Lets make another with interactive and fun to watch.
          </p>
          <a
            class="btn btn-primary"
            href="/add-new-video"
          >
            Lets go!
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
